offline/utils.py

```diff
@@ -0,0 +1,350 @@
+            # of the EEG sample that the cue took place.
+            cue_indices = (np.atleast_1d(cue_times) - offset) * sfreq
+            cue_indices = cue_indices.astype(int)
+
+            # Initialize the label encoder
+            le = LabelEncoder()
+            # Encode the cues using the label encoder
+            cues_encoded = le.fit_transform(cues)
+
+            if with_stim:
+                # Initalize the STIM channel
+                stim = np.zeros(data[0].shape)
+                stim[cue_indices] = cues_encoded + 1
+
+                # Append the STIM channel to the EEG data array
+                data = np.concatenate((data, stim.reshape(1, -1)))
+                # Add stim to the channel types
+                ch_types.append('stim')
+                # Add STIM as a channel
+                ch_names.append('STIM')
+
+    info = mne.create_info(ch_names, sfreq, ch_types)
+
+    # Add the system name to the info
+    info['description'] = 'Unicorn Hybrid Black'
+
+    raw = mne.io.RawArray(data, info)
+
+    # Set the measurement date
+    tmp_dt = datetime.strptime(header['info']['datetime'][0],
+                               "%Y-%m-%dT%H:%M:%S%z")
+    tmp_dt = tmp_dt.astimezone(timezone.utc)
+    raw.set_meas_date(tmp_dt)
+
+    # Create a montage out of the 10-20 system
+    montage = mne.channels.make_standard_montage('standard_1020')
+
+    # Apply the montage
+    raw.set_montage(montage)
+
+    # A line to supress an error message
+    raw._filenames = [file]
+
+    """# Convert time stamps of the Raw object to indices
+    raw_indices = raw.time_as_index(times=raw.times)
+
+    # Raise and error if the cue index is larger than the maximum
+    # index determined by the EEG recording
+    if cue_indices.max() > raw_indices.max():
+        raise Exception(
+            'Cue index is larger than the largest sample index of the Raw object!'
+        )"""
+
+    # Initialize the event array
+    event_arr = np.zeros((len(cue_indices), 3), dtype=int)
+
+    # Store the event information in an array of shape (len(cues), 3)
+    event_arr[:, 0] = cue_indices
+    event_arr[:, 2] = cues_encoded
+
+    # Create a class-encoding correspondence dictionary for the Epochs object
+    event_id = dict(zip(list(le.classes_), range(len(le.classes_))))
+
+    return raw, event_arr, event_id
+
+
+def create_config():
+    """
+    Cofiguration file that creates a pkl file to hold the durations
+    used in the paradigm.
+    """
+
+    exp_durations = {
+            'highlight_length': 0.25,
+            'target_length': 1,
+            'baseline_length': 3,
+            'delay_baseline_arrows': 1,
+            'inter_highlight_length': 0.2,
+            'inter_block_length': 3,
+            'inter_trial_length': 1,
+            'num_highlights': 10
+        }
+
+    with open(r'config_durations.pkl', 'wb') as file:
+        pickle.dump(exp_durations, file)
+    return 0
```

Comment:
Why is 0 returned?

Reply:
I thought it must return to some value, it is removed now.